src/core_plugins/vega_vis/public/vega_vis.vis_type.js

```diff
@@ -0,0 +1,41 @@
+import { VisTypesRegistryProvider } from 'ui/registry/vis_types';
+
+import { VisFactoryProvider } from 'ui/vis/vis_factory';
+import { CATEGORY } from 'ui/vis/vis_category';
+
+import { VegaEditorProvider } from './vega_vis_editor';
+import { VegaRequestHandlerProvider } from './vega_request_handler';
+import './vega_vis.directive';
+import './vega_vis_editor.less';
+
+import defaultSpec from '!!raw-loader!./default.spec.json';
+
+// register the provider with the visTypes registry
+VisTypesRegistryProvider.register(function VegaVisProvider(Private) {
+  const VisFactory = Private(VisFactoryProvider);
+  const VegaEditor = Private(VegaEditorProvider);
+  const vegaRequestHandler = Private(VegaRequestHandlerProvider).handler;
+
+  // return the visType object, which kibana will use to display and configure new
+  // Vis object of this type.
+  return VisFactory.createAngularVisualization({
```

Comment:
I'd prefer to see this rewritten as a base-visualization. So basically, `.createBaseVisualization`. The requirements for angular are very minimal, basically plopping in some content in a node. Also, we are actively removing much of angular outside of Kibana.

So I'd remove the vega-vis directive and pull int the vega_vis_controller directly.